internal/metrics/collectors/manager.go

```diff
@@ -88,12 +110,64 @@ func (nc *LocalManagerMetricsCollector) UpdateLastReloadTime(duration time.Durat
 	nc.lastReloadTime.Set(float64(duration / time.Millisecond))
 }
 
+// UpdateWorkerProcessCount sets the number of NGINX worker processes
+func (nc *LocalManagerMetricsCollector) UpdateWorkerProcessCount(configVersion string) {
+	totalWorkerProcesses, err := getWorkerProcesses()
+	if err != nil {
+		glog.Errorf("Error unable to collect process metrics : %v", err)
+		return
+	}
+	nc.workerProcessTotal.WithLabelValues(configVersion).Set(float64(totalWorkerProcesses))
+	nc.workerProcessTotal.WithLabelValues("old").Set(nc.oldWorkerProcessTotal)
+	nc.oldWorkerProcessTotal = totalWorkerProcesses
+}
+
+func getWorkerProcesses() (float64, error) {
+	var workerProcesses float64
+	procFolders, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return 0, fmt.Errorf("Error in reading directory /proc : %v", err)
+	}
+	var processes []string
+	for _, f := range procFolders {
+		u, err := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+		if err != nil {
+			return 0, fmt.Errorf("Error could not find user for file %v : %v", f.Name(), err)
+		}
+		if u.Name == "nginx user" {
+			processes = append(processes, f.Name())
+		}
+	}
+	for _, file := range processes {
+		statusFile := fmt.Sprintf("/proc/%v/status", file)
+		f, err := os.Open(statusFile)
+		if err != nil {
+			return 0, fmt.Errorf("Error in opening file %v", statusFile)
+		}
+
+		scanner := bufio.NewScanner(f)
+
+		for scanner.Scan() {
+			if strings.Contains(scanner.Text(), "PPid") {
```

Comment:
Nitpick: I think the follow would be clearer due to being more specific and I think it should work the same:
```suggestion
			if strings.HasPrefix(scanner.Text(), "PPid") {
```